client/sites-dashboard/components/sites-display-mode-switcher.tsx

```diff
@@ -11,11 +13,13 @@ const container = css( {
 
 type SitesDisplayMode = 'tile' | 'list';
 
-export const useSitesDisplayMode = () =>
-	useAsyncPreference< SitesDisplayMode >( {
-		defaultValue: 'tile',
+export const useSitesDisplayMode = () => {
+	const siteCount = useSelector( ( state ) => getCurrentUserSiteCount( state ) );
```

Comment:
Not really a fan of adding conditionals to hooks. I think it'd be better if we could specify what the default value should be from the outside through a hook prop instead of hardcoding logic inside it. What if we want to use this hook elsewhere? I think it'd be ideal to pass the `defaultValue` as a prop, on both this hook `useSitesDisplayMode` and `useSitesSorting`.

Reply:
@zaguiini I don't think I'm a huge fan of passing the default value into the hook. Doing so would potentially lead to inconsistent behavior throughout the application (if one caller passed a different default than another).

Given this hook interacts with global state, I think it's fine to have some conditional logic based on the global state.

I agree with your sentiment generally though. I think it'd be applicable to other hooks that aren't dependent on or manipulate some global state.

Reply:
Fair enough, and it's scoped to the sites anyway. The other hook (`useSitesSorting`) is also something we want guarantee streamlining, so I agree, let's keep it as it is 👍